lib/livebook/session.ex

```diff
@@ -2284,12 +2244,15 @@ defmodule Livebook.Session do
   defp before_close(state) do
     maybe_save_notebook_sync(state)
     broadcast_message(state.session_id, :session_closed)
+  end
 
-    if state.data.mode == :app do
-      broadcast_app_message(state.session_id, {:app_terminated, state.session_id})
-    end
+  defp app_report_client_count_change(state) when state.data.mode == :app do
+    client_count = map_size(state.data.clients_map)
+    send(state.app_pid, {:app_client_count_changed, state.session_id, client_count})
```

Comment:
Why are we reporting to the client instead of keeping it in the session and then querying the session? Or is the goal that we never have to query the app?

Reply:
We report it to the app process, which updates the tracker, so when we list apps we can show the current number of clients for each session.

Reply:
Oh, otherwise in a cluster we would need to do a bunch of requests to each session in the cluster, right?! Sounds good to me.

Reply:
Yeah, for listing we want this information in the tracker. And for the apps listing all information we need is in `app.sessions` (and those maps include different information than sessions in the tracker, since we care about different information in case of apps).